Libraries/oneMKL/matrix_mul/matrix_mul_mkl.cpp

```diff
@@ -0,0 +1,185 @@
+//==============================================================
+// Copyright � 2020 Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+//
+// Matrix Multiplication is a simple program that multiplies together two large matrices and verifies the results.
+// This program is implemented using C++ with oneAPI Math Kernel Library (oneMKL)
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <limits>
+#include "mkl.h"
+#include "mkl_blas_sycl.hpp"
+
+using namespace std;
+using namespace sycl;
+
+// Matrix size constants
+auto constexpr size = (600 * 8)  // Must be a multiple of 8.
+auto constexpr M = size / 8
+auto constexpr N = size / 4
+auto constexpr P = size / 2
+
+/**
+ * Perform the matrix multiplication on host to verify results from mkl.
+ */
+int VerifyResult(double *c_back);
+
+int main() {
+  //
+  // Initialize data for Gemm
+  //
+  // C = alpha * op(A) * op(B)  + beta * C
+  //
+  mkl::transpose transA = mkl::transpose::nontrans;
+  mkl::transpose transB = mkl::transpose::nontrans;
+
+  // Matrix data sizes
+  int m = M;
+  int n = P;
+  int k = N;
+
+  // Meading dimensions of data
+  int ldA = m;
+  int ldB = k;
+  int ldC = m;
+
+  // Set scalar fp values
+  double alpha = 1.0;
+  double beta = 0.0;
+
+  // 1D arrays on host side
+
+  auto A = new double[M * N];
+  auto B = new double[N * P];
+  auto C = new double[M * P];
+
+  // Prepare matrix data with column-major style
+  int i, j;
```

Comment:
Declare all the index in the for-loops, even if you repeat it.
This is not Fortran... :-)
Otherwise it is super hard to track the scope...